From 27323ce68d58fcb82a648ab14abbc99acdc36f44 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Wed, 31 Jul 2013 11:01:15 +1000
Subject: [PATCH 2/2] nouveau: add gpu identifier to connector names for
 secondary gpus

this avoids clashes with identifiers from other GPUs.

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
 src/drmmode_display.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 570a9e4..3563b2a 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -1098,6 +1098,10 @@ drmmode_output_init(ScrnInfoPtr pScrn, drmmode_ptr drmmode, int num)
 	if (koutput->connector_type >= NUM_OUTPUT_NAMES)
 		snprintf(name, 32, "Unknown%d-%d", koutput->connector_type,
 			 koutput->connector_type_id);
+	else if (pScrn->is_gpu)
+		snprintf(name, 32, "%s-%d-%d",
+			 output_names[koutput->connector_type], pScrn->scrnIndex - GPU_SCREEN_OFFSET + 1,
+			 koutput->connector_type_id);
 	else
 		snprintf(name, 32, "%s-%d",
 			 output_names[koutput->connector_type],
-- 
1.8.3.1

